osh/cmd_parse.py

```diff
@@ -2183,7 +2193,9 @@ def _ParseAndOr(self):
         break
 
     node = command.AndOr(ops, children)
-    node.spids = op_spids
+    for token in ops:
+        # we cannot use a list comprehension as those are not supported by mycpp
```

Comment:
oh yeah there is a special case for something like

```
tmp = [x for x in mylist if x != 0]
node.spids = tmp
```

Where we turn it into a loop that appends to `tmp`

But yeah the original suggestion might not have worked

This looks fine though